src/main/scala/org/dbpedia/spotlight/wikistats/JsonPediaParser.scala

```diff
@@ -59,5 +72,83 @@ class JsonPediaParser(lang:String)(implicit val sc: SparkContext,implicit val sq
   }
 
 
+  /*
+     Method to Get the list of Surface forms from the wiki
+   */
+  def getSfs() : RDD[String] = {
+
+    dfWikiRDD.select("wid","links.description")
+             .rdd
+             .map(artRow => (artRow.getList[String](1)))
+             .flatMap(sf => sf)
+  }
+
+  /*
+  Method to get the wid and article text from the wiki dump
+   */
+  //TODO Need to improve the filter condition of blank articles. we can include in the where clause of dataframe
+  def getArticleText(): RDD[(Long,String)] = {
+
+    dfWikiRDD.select("wid","wikiText")
+             .rdd
+             .map(artRow => {
+                 (artRow.getLong(0),artRow.getString(1))
+             })
+             .filter(artRow => artRow._2.length > 0)
+  }
+
+  /*
+   Logic to Get Surface Forms and URIs from the wikiDump
+   */
+  def getSfURI(): RDD[(Long,String,String)]= {
+
+    val sfUriRDD = dfWikiRDD.select("wid","links.description","links.id")
+             .rdd
+             .map(artRow => (artRow.getLong(0),artRow.getList[String](1),artRow.getList[String](2)))
+             .map{case (wid,sfArray,uriArray) => (wid,sfArray.zip(uriArray))}
+             .flatMap{case (wid,uriSf) => for(x <- uriSf) yield (wid,x._1,x._2)}
+    sfUriRDD
+
+  }
+  /*
+
+  Logic to Create Memory Token Store
+   */
+  def createTokenTypeStore(tokenTypes:List[TokenType]): MemoryTokenTypeStore =  {
+
+    val tokenTypeStore = new MemoryTokenTypeStore()
+    val tokens = new Array[String](tokenTypes.size + 1)
+    val counts = new Array[Int](tokenTypes.size + 1)
+
+    tokenTypes.map(token => {
+      tokens(token.id) = token.tokenType
+      counts(token.id) = token.count
+
+    })
+
+    tokenTypeStore.tokenForId  = tokens.array
+    tokenTypeStore.counts = counts.array
+    tokenTypeStore.loaded()
+
+    return tokenTypeStore
+  }
+
+  /*
+  Logic for building the memory type tokens
+   */
+  def getTokens(): List[TokenType] ={
+
+    val stemmer = new Stemmer()
+    val locale = new Locale(lang)
+    val lst = new LanguageIndependentStringTokenizer(locale, stemmer)
+    //Below Logic is for creating Token Store from the Surface forms
+    //TODO Getting Searlization error Hence Using Collect and ToList. May need to change in Future
```

Comment:
Could you add this as an Issue to the repo.
In my experience all of these `ToDo` get forgotten.